server/src/main/java/org/elasticsearch/gateway/ReplicaShardAllocator.java

```diff
@@ -99,11 +101,14 @@ public void processExistingRecoveries(RoutingAllocation allocation) {
                                 currentNode, nodeWithHighestMatch);
                         final Set<String> failedNodeIds =
                             shard.unassignedInfo() == null ? Collections.emptySet() : shard.unassignedInfo().getFailedNodeIds();
+                        boolean nodeIsRestarting = Optional.ofNullable(metadata.nodeShutdowns().get(shard.currentNodeId()))
+                            .map(shutdownInfo -> shutdownInfo.getType().equals(SingleNodeShutdownMetadata.Type.RESTART))
+                            .orElse(false);
                         UnassignedInfo unassignedInfo = new UnassignedInfo(UnassignedInfo.Reason.REALLOCATED_REPLICA,
                             "existing allocation of replica to [" + currentNode + "] cancelled, can perform a noop recovery on ["+
                                 nodeWithHighestMatch + "]",
                             null, 0, allocation.getCurrentNanoTime(), System.currentTimeMillis(), false,
-                            UnassignedInfo.AllocationStatus.NO_ATTEMPT, failedNodeIds);
+                            UnassignedInfo.AllocationStatus.NO_ATTEMPT, failedNodeIds, null, nodeIsRestarting);
```

Comment:
I think we should just pass in false for restarting here?

Also, I wonder if we should for completeness pass in the `currentNodeId` for `lastAllocatedNodeId`? Not that it truly matters I think.

Reply:
To the first point, this boolean no longer exists.

To the second, we cannot due so due to the assertion recommended above that `lastAllocatedNodeId` only be populated when the node has left (and here it has a different reason).